src/main/groovy/org/transmartproject/batch/tag/TagTypeWriter.groovy

```diff
@@ -0,0 +1,24 @@
+package org.transmartproject.batch.tag
+
+import groovy.util.logging.Slf4j
+import org.springframework.batch.item.ItemWriter
+import org.springframework.beans.factory.annotation.Autowired
+
+/**
+ * Database writer of tag types
+ */
+@Slf4j
+class TagTypeWriter implements ItemWriter<TagType> {
+
+    @Autowired
+    InsertTagTypeService insertTagTypeService
+
+
+    @Override
+    void write(List<? extends TagType> items) throws Exception {
+        insertTagTypeService.deleteAllTagsForTagTypes()
+        insertTagTypeService.deleteAllTagTypes()
```

Comment:
@gijskant The cleanup of the tables in this place doesn't look right. The `write` method would be executed multiple times (for each chunk) during the upload. At the end you would get only items written down at the last chunk.